flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartition.java

```diff
@@ -149,7 +149,11 @@ private void writeAndCloseBufferConsumer(BufferConsumer bufferConsumer) throws I
 		try {
 			final Buffer buffer = bufferConsumer.build();
 			try {
-				data.writeBuffer(buffer);
+				if (bufferCompressor != null && buffer.isBuffer() && buffer.readableBytes() > 0) {
```

Comment:
it is better to give some comments here why we only compress buffer, not for event.